src/service/helperFunctions.tsx

```diff
@@ -146,3 +146,41 @@ export const findUniqueSelector = (
     "*"
   );
 };
+
+export const isStepInActionProcess = (step: Step): boolean => {
+  if (
+    (step.action === StepAction.RECORD_CLICKS_KEYS && !step.recordedClicksAndKeys!.length) ||
+    (step.action !== StepAction.RECORD_CLICKS_KEYS && !step.tagName)
+  ) {
+    return true
+  } else {
+    return false
+  }
+}
+
+export const reduceInProcessSteps = (steps: Step[]): Step[] => {
+  const reducedSteps = []
+  let canHaveMoreProcessSteps = true
+  for (const step of steps) {
+    if (isStepInActionProcess(step)) {
+      if (canHaveMoreProcessSteps) {
+        reducedSteps.push(step)
+        canHaveMoreProcessSteps = false
+      }
+    } else {
+      reducedSteps.push(step)
+    }
+  }
+
+  return reducedSteps
+}
+
+// for future use
+export const stepAlreadyInProcess = (steps: Step[]): boolean => {
+  const inProcessSteps = steps.filter(step => isStepInActionProcess(step))
+  if (inProcessSteps.length) {
+    return true
+  } else {
+    return false
+  }
+}
```

Comment:
You can remove `reduceInProcessSteps` & `stepAlreadyInProcess`. Last PR is forbidding having multiple "empty" steps